cachi2/core/package_managers/cargo/main.py

```diff
@@ -22,20 +21,29 @@ def fetch_cargo_source(request: Request) -> RequestOutput:
 
     for package in request.cargo_packages:
         package_dir = request.source_dir.join_within_root(package.path)
-        components.extend(_resolve_cargo_package(package_dir, request.output_dir))
         # cargo allows to specify configuration per-package
         # https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure
-        project_files.append(_use_vendored_sources(package_dir))
+        fetched_components, cfg_template = _resolve_cargo_package(package_dir, request.output_dir)
+        components.extend(fetched_components)
+        tail = 'directory = "${output_dir}/deps/cargo"'
+        # absolute path has to be replaced with relative path for sources relocation to work:
+        # (absolute path also happens to be the last line generated by cargo)
```

Comment:
Can this path mangling be incorporated in the resolve method and return a final template product? I mean the resolve method has all the necessary information.

Reply:
>\# absolute path has to be replaced with relative path for sources relocation to work:
>\# (absolute path also happens to be the last line generated by cargo)

This is very error prone I'm afraid, we're relying on cargo's CLI output ordering which in general isn't considered part of the public API, so we'll need to be more thorough with the check (`startswith` should be sufficient, `re` would IMO be overkill for such a simple replacement).